src/libraries/System.Private.CoreLib/src/System/Boolean.cs

```diff
@@ -110,7 +117,14 @@ public bool TryFormat(Span<char> destination, out int charsWritten)
             {
                 if ((uint)destination.Length > 4)
                 {
-                    BinaryPrimitives.WriteUInt64LittleEndian(MemoryMarshal.AsBytes(destination), 0x73006C00610046); // "Fals"
+                    if (BitConverter.IsLittleEndian)
+                    {
+                        BinaryPrimitives.WriteUInt64LittleEndian(MemoryMarshal.AsBytes(destination), 0x73006C00610046); // "Fals"
+                    }
+                    else
+                    {
+                        BinaryPrimitives.WriteUInt64BigEndian(MemoryMarshal.AsBytes(destination), 0x460061006C0073); // "Fals"
+                    }
```

Comment:
I don't understand, wasn't `.WriteUInt64LittleEndian` supposed to handle endianess via ReverseEndianness() for value?
```

        /// <summary>
        /// Write a UInt64 into a span of bytes as little endian.
        /// </summary>
        [CLSCompliant(false)]
        [MethodImpl(MethodImplOptions.AggressiveInlining)]
        public static void WriteUInt64LittleEndian(Span<byte> destination, ulong value)
        {
            if (!BitConverter.IsLittleEndian)
            {
                value = ReverseEndianness(value);
            }
            MemoryMarshal.Write(destination, ref value);
        }
```

Reply:
if it wasn't then you can re-write code to just `MemoryMarshal.Write(destination, 0x73006C00610046);` I guess

Reply:
The problem is that you need to distinguish between swapping the order of the *characters* and the order of the *bytes* within one two-byte character.   See https://github.com/dotnet/runtime/pull/64782#issuecomment-1033785023 for an example.  There's no single store that does the correct thing for both of these simultaneously on both big- and little-endian platforms.

Reply:
I see, thanks for explanation! anyway maybe it's still better to use 
```
MemoryMarshal.Write(destination, IsLittleEndian ? 0x73006C00610046 : ...)
```
? 
less verbose IMO

Reply:
That would work as well.   I can try this out if you prefer.

Reply:
Patch updated accordingly.